### PR TITLE
Ultra-LRT

### PR DESCRIPTION
**Pull request by maruf0011**
- [x] Initial implementation of Vault and Delegator
- [x] Individual test for Vault and Delegator
- [x] Integration of Vault and Delegator
- [x] Test Vault and Delegator integration and withdrawal
- [ ] Comment & resolve TODO's
- [ ] Cleanup 
- [ ] Deploy Script

### CODE REVIEW

File: src/vaults/restaking/WithdrawalEscrowV2.sol [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        resolvingEpoch += 1;
+    }
+
+    /**
+     * @notice Redeem withdrawal request
+     * @param user address
+     * @param epoch withdrawal request epoch
+     * @return received assets
+     */
+    function redeem(address user, uint256 epoch) external returns (uint256) {
```
Comment (Omurshid) [inline src/vaults/restaking/WithdrawalEscrowV2.sol:112]: maybe allow 1 tx for redeeming all user's pending withdrawals

File: src/vaults/restaking/UltraLRTStorage.sol [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    // delegator array
+    IDelegator[MAX_DELEGATOR] public delegatorQueue;
+    mapping(address => DelegatorInfo) public delegatorMap;
+
+    //active delegator count
+    uint256 public delegatorCount;
+
+    modifier whenDepositNotPaused() {
+        require(depositPaused == 0, "Deposit Paused.");
```
Comment (Omurshid) [inline src/vaults/restaking/UltraLRTStorage.sol:60]: When is depositPaused set to not 0?

File: src/vaults/restaking/WithdrawalEscrowV2.sol (hunk 1 above)
Comment (maruf0011) [reply src/vaults/restaking/WithdrawalEscrowV2.sol]: added `redeemMultiEpoch` function to redeem multiple epoch in single tx.

File: src/vaults/restaking/UltraLRTStorage.sol (hunk 2 above)
Comment (maruf0011) [reply src/vaults/restaking/UltraLRTStorage.sol:60]: Added :), Thanks for the catch.

**Pull request closed without merging**